### PR TITLE
Add some improvements

### PR DESCRIPTION
**Pull request by antca**
Summary:

- Removes the ability for the components to specify options.
  The Issue was that any options set this way has side effects on other components subscribing to this path.
  If this behavior is really needed, you can still access the `flux` in the `@store` callback.
- Add the ability to set options on a whole store or on a specific path via `HTTP.store#options`.
- Removes the `refreshInterval` option. This feature can be easily achieved by the user dirrectly with better flexibility.
- Clean the public API. Now you can only interact with stores via `query` but not via `path` anymore.
- Add `rewritePath` option taking a function to rewrite the path just before sending the `fetch` request.
- Removes specific actions (HTTP/Memory), generic Actions are more flexible.

Modifications:

- Fix HTTP/Store#observe to allow HTTP to work without SSR
- Change bindings parameters handling
- Pass flux instead of context to actions getBindings
- Remove binding parameters and unify public Store method parameters
- Add options to HTTP/Store
- Fix HTTPStore test
- Add a rewritePath option
- Remove specific actions types
- Pass flux when dispatching actions

### CODE REVIEW

File: lib/HTTPStore/index.js [hunk 1]
```diff
@@ -74,15 +76,12 @@ class HTTPStore extends Store {
...
+   * @param {Object} query Query to retrieve the matching State.
+   * @return {State} Matching State.
    */
-  readFromState(queryOrPath) {
-    const path = typeof queryOrPath === 'string' ? queryOrPath : this.toPath(queryOrPath);
-    if(!this.data.has(path)) {
-      return Store.State.pending({ path });
-    }
-    return this.data.get(path).state;
+  readFromState(query) {
```
Comment (eTorAken) [inline lib/HTTPStore/index.js]: :+1:

File: lib/__tests__/preparable.js [hunk 2]
```diff
@@ -44,7 +44,8 @@ describe('preparable', () => {
     })
     class Test extends React.Component {}
     await Test[$prepare]();
-    should(spy).have.property('callCount').which.is.exactly(2);
+    const expectedCallCount = 2;
```
Comment (eTorAken) [inline lib/__tests__/preparable.js]: `EXPECTED_CALL_COUNT`

File: lib/HTTPStore/index.js [hunk 3]
```diff
@@ -200,16 +210,40 @@ class HTTPStore extends Store {
...
+    return Object.assign({}, this._options, path ? this._findOrCreatePathData(path).options : {});
+  }
+
+  /**
+   * Updates options
+   * @param  {Object|null} [query] Query of the query specific options.
+   * @param  {Object} options Options to merge with the current options.
+   * @return {HTTPStore} Instance of the current HTTPStore.
+   */
+  options(query, options) {
```
Comment (antca) [inline lib/HTTPStore/index.js]: I am not completely sure that allowing to set options per query is a good idea after all. It adds a little overhead with `_getOptions` and I can't remember a legit usage. What do you think @elierotenberg ?

**Comment by Brykou** [comment]
Ok pour moi.

**Pull request merged**